extern/imgui_patched/imgui.cpp

```diff
@@ -9570,6 +9574,8 @@ void ImGui::UpdateInputEvents(bool trickle_fast_inputs)
         }
         else if (e->Type == ImGuiInputEventType_Text)
         {
+            bool trickle_fast_inputs = true; // forcing trickle for Text event
```

Comment:
I will question this one. What will happen if the key repeat rate is higher than the frame rate?

Reply:
My understanding is that the `Text` trickle mechanism here only activates if there's a mixture of `Key` events with the `Text` ones (the `key_changed` flag generated by a 'Key' event gates it). Pressing a key only generates one `Key` event down and one `Key` event up when it releases, and the auto-repetitions only generate `Text` events, which by themselves will not cause the event trickle, and instead just add to `InputQueueCharacters`.

So, `Key` events should be sparse, like `MouseButton` events. `Text` events might be as rapid as the auto-repeat mechanism allows, but don't cause the trickle delay on their own.

Reply:
You sure? From what I know, SDL may generate several key down effects (with a repeat flag set).

Reply:
Yes. I did test it to verify before saying that.

SDL does generate repeat KEYDOWN events, but ImGui filters those out. See: [ImGuiIO::AddKeyAnalogEvent](https://github.com/tildearrow/furnace/blob/0444588cca15dd08bd61060fb474637f1360d207/extern/imgui_patched/imgui.cpp#L1447)

Because of this KEYDOWN repeat does not spam the ImGui event queue, and cannot cause trickle.

Reply:
I just tested.

Sorry, but with a key repeat rate of 50 per second:
- Furnace running at 60fps: Stutter, every 5 frames. Unacceptable.
- Furnace running at 30fps: Mouse lag when moving window.

Reply:
I'll respond below.